src/test/java/org/opensearch/geospatial/ip2geo/action/RestGetDatasourceHandlerTests.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.geospatial.ip2geo.action;
+
+import static org.opensearch.geospatial.shared.URLBuilder.URL_DELIMITER;
+import static org.opensearch.geospatial.shared.URLBuilder.getPluginURLPrefix;
+
+import java.util.Locale;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.apache.commons.lang3.StringUtils;
+import org.junit.Before;
+import org.opensearch.geospatial.GeospatialTestHelper;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.test.rest.FakeRestRequest;
+import org.opensearch.test.rest.RestActionTestCase;
+
+public class RestGetDatasourceHandlerTests extends RestActionTestCase {
+    private String PATH_FOR_ALL = String.join(URL_DELIMITER, getPluginURLPrefix(), "ip2geo/datasource");
+    private String path;
+    private RestGetDatasourceHandler action;
+
+    @Before
+    public void setupAction() {
+        action = new RestGetDatasourceHandler();
+        controller().registerHandler(action);
+        path = String.join(URL_DELIMITER, getPluginURLPrefix(), "ip2geo/datasource/%s");
+    }
+
+    public void testPrepareRequest() {
```

Comment:
Given the number of tests and things that are being tested in these tests, can we rename the tests to a pattern like this:

testABC_whenPQR_thenXYZ.

Example:
public void testPrepareRequest_whenValidInput_thenSuccess

This will do the benefit where just by reading the name of the function we can get to know what we is being tested by this function. It will be good for debugging too to pin point just by seeing tests failure what is the problem.

We are using similar pattern in K-NN repo. May you want to refer from there.

Reply:
@heemin32 so are you making the change?

Reply:
Will change it gradually but not everything in this PR.